### PR TITLE
disable continue/submit button if section of form not complete

### PR DESCRIPTION
**Pull request by dnoneill**
Screenshot 2024-05-01 at 4 08 26 PM
Screenshot 2024-05-01 at 4 08 14 PM

### CODE REVIEW

**Comment by cbeer** [comment]
I think there's a problem when choosing pickup request. The submit button is disabled even though there's no form fields to fill out?

File: app/javascript/controllers/patron_request_controller.js [hunk 1]
```diff
@@ -8,9 +8,34 @@ export default class extends Controller {
...
+    return Array.from(accordion.querySelectorAll('[required], [data-required]')).find(x => formData.getAll(x.name).every(x => !x))
+  }
+
+  nextStepButtonEventListeners() {
+    var _this = this;
+    this.accordionTargets.forEach(accord => {
+      _this.enableNextButton(accord)
+      accord.addEventListener("input", function() {
+        _this.enableNextButton(accord)
+      });
```
Comment (cbeer) [inline app/javascript/controllers/patron_request_controller.js:26]: Can we add this event listener the stimulus way? (e.g.  something like`data-action="input->patronRequest#checkIfTheRequiredFieldsAreFilledOutAndEnableTheNextButton"`)

Comment (dnoneill) [reply app/javascript/controllers/patron_request_controller.js]: Sounds good. Pushed

**Comment by dnoneill** [comment]
closing in favor of https://github.com/sul-dlss/sul-requests/pull/2389

**Pull request closed without merging**